### PR TITLE
Less repeated properties.

### PR DESCRIPTION
**Pull request by moruzerinho6**
> Zerinho: "Pra que você ta dando await no send se você não ta guardando a mensagem"
> Phaze: "Pq com rosa fica mais rápido"

I think david took the notes from phaze with the double await, lmao.

This PR creates variables to reduce the use of repeated properties in many lines related to league of legends commands.

### CODE REVIEW

File: src/commands/games/leagueoflegends/champion.js [hunk 1]
```diff
@@ -1,45 +1,49 @@
...
+      }]
+    }, client)
+  }
+ 
+  async run ({ t, author, channel, prefix, language }, champion) {
+    channel.startTyping()
+    const embed = new SwitchbladeEmbed(author)
+    try {
+      const lolApi = this.client.apis.lol
+      const champ = await lolApi.fetchChampion(champion, language)
```
Comment (fernando457829) [inline src/commands/games/leagueoflegends/champion.js]: Destruct it too?

```suggestion
      const { stats, name, title, image, blurb, spells, skins } = await lolApi.fetchChampion(champion, language)
```

File: src/commands/games/leagueoflegends/champion.js [hunk 2]
```diff
@@ -1,45 +1,49 @@
...
+        type: 'string', full: true, missingError: 'commands:leagueoflegends.subcommands.champion.noChampion'
+      }]
+    }, client)
+  }
+ 
+  async run ({ t, author, channel, prefix, language }, champion) {
+    channel.startTyping()
+    const embed = new SwitchbladeEmbed(author)
+    try {
+      const lolApi = this.client.apis.lol
```
Comment (fernando457829) [inline src/commands/games/leagueoflegends/champion.js]: What about destruct `lolApi`?

```suggestion
      const { fetchChampion, getLocale } = this.client.apis.lol
```

File: src/commands/games/leagueoflegends/rotation.js [hunk 3]
```diff
@@ -1,34 +1,36 @@
...
+    }, client)
+  }
+
+  async run ({ t, author, channel, language, flags }) {
+    channel.startTyping()
+    const embed = new SwitchbladeEmbed(author)
+    const parentCommand = this.parentCommand
+    const lolApi = this.client.apis.lol
+    try {
+      const rotation = await lolApi.fetchChampionRotation()
```
Comment (fernando457829) [inline src/commands/games/leagueoflegends/rotation.js]: And destruct too

File: src/commands/games/leagueoflegends/rotation.js [hunk 4]
```diff
@@ -1,34 +1,36 @@
...
+        type: 'booleanFlag', name: 'newplayers', aliases: ['np']
+      }]]
+    }, client)
+  }
+
+  async run ({ t, author, channel, language, flags }) {
+    channel.startTyping()
+    const embed = new SwitchbladeEmbed(author)
+    const parentCommand = this.parentCommand
+    const lolApi = this.client.apis.lol
```
Comment (fernando457829) [inline src/commands/games/leagueoflegends/rotation.js]: Destruct too

File: src/commands/games/leagueoflegends/skin.js [hunk 5]
```diff
@@ -1,30 +1,31 @@
...
+        type: 'string', full: true, missingError: 'commands:leagueoflegends.subcommands.skin.noSkin'
+      }]
+    }, client)
+  }
+
+  async run ({ t, author, channel, language }, skin) {
+    channel.startTyping()
+    const embed = new SwitchbladeEmbed(author)
+    try {
+      const skinInfo = await this.client.apis.lol.fetchSkin(skin, this.client)
```
Comment (fernando457829) [inline src/commands/games/leagueoflegends/skin.js]: More destruct?

File: src/commands/games/leagueoflegends/champion.js [hunk 6]
```diff
@@ -1,45 +1,50 @@
...
+    }, client)
+  }
+ 
+  async run ({ t, author, channel, prefix, language }, champion) {
+    channel.startTyping()
+    const embed = new SwitchbladeEmbed(author)
+    try {
+      const { fetchChampion, getLocale, version } = this.client.apis.lol
+      const champFetch = await fetchChampion(champion, language)
+      const { name, title, blurb, image, spells, skins } = champFetch
```
Comment (fernando457829) [inline src/commands/games/leagueoflegends/champion.js]: WTF?
no please

```suggestion
      const { name, title, blurb, image, spells, skins, stats } = await fetchChampion(champion, language)
```

File: src/commands/games/leagueoflegends/champion.js [hunk 7]
```diff
@@ -1,45 +1,50 @@
...
+      const { hp, hpregen, mp, mpregen, armor, attackdamage, crit, hpperlevel, hpregenperlevel, mpperlevel, mpregenperlevel, attackdamageperlevel, critperlevel } = champFetch.stats
```
Comment (fernando457829) [inline src/commands/games/leagueoflegends/champion.js]:
```suggestion
      const { hp, hpregen, mp, mpregen, armor, attackdamage, crit, hpperlevel, hpregenperlevel, mpperlevel, mpregenperlevel, attackdamageperlevel, critperlevel } = stats
```

File: src/commands/games/leagueoflegends/champion.js [hunk 8]
```diff
@@ -1,45 +1,50 @@
...
+      const locale = await getLocale(language)
```
Comment (fernando457829) [inline src/commands/games/leagueoflegends/champion.js]:
```suggestion
      const { Health, HealthRegen, Mana, ManaRegen, Armor, Attack, CriticalStrike, level } = await getLocale(language)
```

File: src/commands/games/leagueoflegends/rotation.js [hunk 9]
```diff
@@ -1,34 +1,36 @@
...
+  async run ({ t, author, channel, language, flags }) {
+    channel.startTyping()
+    const embed = new SwitchbladeEmbed(author)
+    const { embedColor, authorString, authorImage, authorURL } = this.parentCommand
+    const { fetchChampionRotation, fetchChampion } = this.client.apis.lol
+    try {
+      const {freeChampionIdsForNewPlayers, freeChampionIds} = await fetchChampionRotation()
+      const champions = flags['newplayers'] ? freeChampionIdsForNewPlayers : freeChampionIds
+      const championPayload = await Promise.all(champions.map(async c => {
+        var payload = await fetchChampion(c, language, true)
```
Comment (fernando457829) [inline src/commands/games/leagueoflegends/rotation.js]: use const and destruct!

```suggestion
        const { name, title } = await fetchChampion(c, language, true)
```

File: src/commands/games/leagueoflegends/rotation.js [hunk 10]
```diff
@@ -1,34 +1,36 @@
...
+  }
+
+  async run ({ t, author, channel, language, flags }) {
+    channel.startTyping()
+    const embed = new SwitchbladeEmbed(author)
+    const { embedColor, authorString, authorImage, authorURL } = this.parentCommand
+    const { fetchChampionRotation, fetchChampion } = this.client.apis.lol
+    try {
+      const {freeChampionIdsForNewPlayers, freeChampionIds} = await fetchChampionRotation()
+      const champions = flags['newplayers'] ? freeChampionIdsForNewPlayers : freeChampionIds
```
Comment (fernando457829) [inline src/commands/games/leagueoflegends/rotation.js]: use dot notation

```suggestion
      const champions = flags.newplayers ? freeChampionIdsForNewPlayers : freeChampionIds
```

File: src/commands/games/leagueoflegends/rotation.js [hunk 11]
```diff
@@ -1,34 +1,36 @@
...
+    }, client)
+  }
+
+  async run ({ t, author, channel, language, flags }) {
+    channel.startTyping()
+    const embed = new SwitchbladeEmbed(author)
+    const { embedColor, authorString, authorImage, authorURL } = this.parentCommand
+    const { fetchChampionRotation, fetchChampion } = this.client.apis.lol
+    try {
+      const {freeChampionIdsForNewPlayers, freeChampionIds} = await fetchChampionRotation()
```
Comment (fernando457829) [inline src/commands/games/leagueoflegends/rotation.js]: Add space on brackets

```suggestion
      const { freeChampionIdsForNewPlayers, freeChampionIds } = await fetchChampionRotation()
```

File: src/commands/games/leagueoflegends/champion.js [hunk 12]
```diff
@@ -1,45 +1,50 @@
...
+      const locLevel = locale.Level
+      const {embedColor, authorString, authorImage, authorURL} = this.parentCommand
```
Comment (fernando457829) [inline src/commands/games/leagueoflegends/champion.js]: Add space on brackets
```suggestion
      const { embedColor, authorString, authorImage, authorURL } = this.parentCommand
```

File: src/commands/games/leagueoflegends/skin.js [hunk 13]
```diff
@@ -1,30 +1,31 @@
...
+      }]
+    }, client)
+  }
+
+  async run ({ t, author, channel, language }, skin) {
+    channel.startTyping()
+    const embed = new SwitchbladeEmbed(author)
+    try {
+      const { name, splashUrl, videoUrl } = await this.client.apis.lol.fetchSkin(skin, this.client)
+      const { embedColor, authorString, authorImage, authorURL} = this.parentCommand
```
Comment (fernando457829) [inline src/commands/games/leagueoflegends/skin.js]:
```suggestion
      const { embedColor, authorString, authorImage, authorURL } = this.parentCommand
```

File: src/commands/games/leagueoflegends/skin.js [hunk 14]
```diff
@@ -1,30 +1,31 @@
...
+      embed.setColor(embedColor)
+        .setAuthor(t(authorString), authorImage, authorURL)
+        .setTitle(`${name}`)
```
Comment (fernando457829) [inline src/commands/games/leagueoflegends/skin.js]: Remove literal string

```suggestion
        .setTitle(name)
```

File: src/commands/games/leagueoflegends/skin.js [hunk 15]
```diff
@@ -1,30 +1,31 @@
...
+        .setImage(splashUrl)
+        .setDescription(t('commands:leagueoflegends.subcommands.skin.description', { videoUrl: videoUrl }))
```
Comment (fernando457829) [inline src/commands/games/leagueoflegends/skin.js]: Use shorthand notation

```suggestion
        .setDescription(t('commands:leagueoflegends.subcommands.skin.description', { videoUrl }))
```

File: src/commands/games/leagueoflegends/champion.js (hunk 6 above)
Comment (moruzerinho6) [reply src/commands/games/leagueoflegends/champion.js]: The reason I did this was to not make 2 api calls, but yeah, your solution is better.

File: src/commands/games/leagueoflegends/champion.js (hunk 8 above)
Comment (moruzerinho6) [reply src/commands/games/leagueoflegends/champion.js]: It can't be ``level``, it needs to be ``Level``.

Comment (fernando457829) [reply src/commands/games/leagueoflegends/champion.js]: yeah, I wrote wrong lol

**Comment by fernando457829** [comment]
LGTM

**Pull request merged**